### PR TITLE
RFC - add support for node 4.2.2

### PR DESCRIPTION
**Pull request by lholmquist**
For https://issues.jboss.org/browse/AGPUSH-1541,  i needed to update the grunt-uglify task for everything to work.

We should also update the other dependecies to the latest.

Perhaps in the same PR.   I also added node 0.12 and 4.2.2 to travis, so we will see if everthting passes first  :).  also removed 0.8 from travis

### CODE REVIEW

File: admin-ui/.travis.yml [hunk 1]
```diff
@@ -1,7 +1,8 @@
 language: node_js
 node_js:
-  - '0.8'
   - '0.10'
+  - '0.12'
+  - '4.2.2'
```
Comment (matzew) [inline admin-ui/.travis.yml]: we need to keep (or should) 0.10 - because we used it a very long time ?

**Comment by matzew** [comment]
Don't we need a POM.xml update, @lholmquist ?

**Comment by lholmquist** [comment]
oh ffs,  probably  :)

**Comment by matzew** [comment]
From the travis log:
```
[INFO] Installing node version v0.10.31
[INFO] Creating temporary directory /home/travis/build/aerogear/aerogear-unifiedpush-server/admin-ui/node_tmp
[INFO] Downloading Node.js from http://nodejs.org/dist/v0.10.31/node-v0.10.31-linux-x64.tar.gz to /home/travis/build/aerogear/aerogear-unifiedpush-server/admin-ui/node_tmp/node.tar.gz
[INFO] No proxies configured
[INFO] No proxy was configured, downloading directly
[INFO] Extracting Node.js files in node_tmp
[INFO] Unpacking /home/travis/build/aerogear/aerogear-unifiedpush-server/admin-ui/node_tmp/node.tar.gz into /home/travis/build/aerogear/aerogear-unifiedpush-server/admin-ui/node_tmp
[INFO] Moving node binary to /home/travis/build/aerogear/aerogear-unifiedpush-server/admin-ui/node/node
[INFO] Deleting temporary directory /home/travis/build/aerogear/aerogear-unifiedpush-server/admin-ui/node_tmp
[INFO] Installed node locally.
[INFO] Installing npm version 1.4.23
```

**Comment by lholmquist** [comment]
@matzew yea,  changed it locally, testing now

**Comment by matzew** [comment]
https://github.com/aerogear/aerogear-unifiedpush-server/blob/master/admin-ui/pom.xml#L60-L61

**Comment by lholmquist** [comment]
@matzew @lfryc mind taking a look.  i think i got all the pieces and parts.

might have to do a `npm cache clear` before hand

I updated all the dependecies to the latest also

**Comment by lfryc** [comment]
Tested:
* mvn clean install -Dfrontend.clean.force
* with node=4.2.2:
  * grunt dist
  * grunt server
  * grunt jbosswebDist

````
$ ✗ npm outdated
Package  Current  Wanted  Latest  Location
tmp       0.0.23  0.0.23  0.0.28  tmp
````
(one "outdated" dep, but that's fine)

Nice, I :heart: it, @lholmquist 

:+1: Merging...

**Comment by lfryc** [comment]
@matzew do you want to take a look before I merge it?

**Comment by matzew** [comment]
works for me (tm)

:+1:

**Pull request merged**